src/go/rpk/pkg/cli/cmd/group/offset_delete.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package group
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+)
+
+func NewOffsetDeleteCommand(fs afero.Fs) *cobra.Command {
+	var (
+		fromFile        string
+		topicPartitions []string
+	)
+	cmd := &cobra.Command{
+		Use:   "offset-delete [GROUP] --from-file FILE --topic foo:0,1,2",
+		Short: "Delete offsets for a kafka group",
+		Long: `Forcefully delete offsets for a kafka group.
+
+The broker will only allow the request to succeed if the group is in a dead
+state (no subscriptions) or there are no subscriptions for offsets for
+topic/partitions requested to be deleted.
+
+Use either the --from-file or the --topic option. They are mututally exclusive.
+To indicate which topics or topic partitions you'd like to remove offsets from use
+the --topic (-t) flag, followed by a comma separated list of partition ids. Supplying
+no list will delete all offsets for all partitions for a given topic.
+
+You may also provide a text file to indicate topic/partition tuples. Use the
+--from-file flag for this option. The file must contain lines of topic/partitions
+separated by a tab or space. Example:
+
+topic_a 0
+topic_a 1
+topic_b 0
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			// Parse input from file or command line
+			var topicsSet kadm.TopicsSet
+			if len(topicPartitions) > 0 {
+				topicsSet, err = parseTopicPartitionsArgs(topicPartitions)
+			} else if fromFile != "" {
```

Comment:
Above this, can we have a conditional that exits if len(topicPartitions) > 0 && fromFile != ""?

Reply:
oh nice, I missed that